### DISCUSSION TITLE
Filtering talks by language in schedule

### DISCUSSION THREAD

**Discussion by MacLemon** [pretalx/pretalx#1202] [Ideas]
While running PrivacyWeek 2021 we've come across an interesting question by our attendees multiple times. So we hope it's warrented to suggest this feature idea.

People wanted to filter the talk schedule by source language. I say source language because we provide simultaneous interpretation. People wanted to see a list of talks in their native language but not the ones they'd only be able to follow via the translated audio track. (We provide interpretation en ←→ de.)
The language information is already available in the event metadata, but it's only displayed in a talk's details, not in the schedule or overviews, changelog, etc.

As an additional language we also were able to provide a few talks in sign language. We translate almost all talks in spoken languages, but only had a select handful in sign language. So being able to filter by sign-language would be tremendously helpful to the people who depend on it.

Sign-language is a separate language, just like spoken english, german or elvish. Of course there's differences among sign languages between for example Austria and Germany, so it may also be relevant to specify the exact kind of signing provided.

Technically, this seems to be two arrays of languages, one for 🗣 spoken languages, and and additional one for 🙌  signed languages.

I'll try to extend thoughts on UI after our conference.

**Comment by ruru4143** [comment]
[Mobilizon](https://joinmobilizon.org/en/) is using this [icon](https://framagit.org/framasoft/mobilizon/-/blob/master/js/public/img/sign_language_monochrome.svg):

![sign language](https://framagit.org/framasoft/mobilizon/-/raw/master/js/public/img/sign_language_monochrome.svg)

**Comment by rashfael** [comment]
Font Awesome also has the two thumb-index-finger hands https://fontawesome.com/v6.0/icons/hands-asl-interpreting.
For filtering I would favor a single, compact search/filter input field which reveals additional filter options on interaction, to save on screen space (especially on mobile).
I think we should also show language/translation/sign language information on each session card, because (in my opinion) list items should show the atttributes which they will be filtered by. Adding all languages in text won't fit though, so a compromise could be "short code for source language" + icon for translation + icon for signed language with lists of translations and signing on interaction with the icons, which would indicate "has translated audio" and "has interpreted sign language" at a glance and which languages exactly on interation? Or make the icons non-interactive and show the exact languages on the detail page.

**Comment by MacLemon** [comment]
I took the liberty to ask around on the Mastodon part of the [Fediverse](https://en.wikipedia.org/wiki/Fediverse) [in german language](https://chaos.social/@MacLemon/107240898809720755) and [in english](https://chaos.social/@MacLemon/107240885206615419) on what sign-language users would actually like to see and what would likely be universally understood.

As in every seemingly small problem lies a larger one at hand that's just waiting to be uncovered. This topic is no different. 

In my personal conclusion of the answers I have received in direct and public responses:
- An Icon like the [👐 (Open Hands U+1F450)](https://emojipedia.org/open-hands/) Emoji or the also mentioned [`thumb-index-finger hands`](https://fontawesome.com/v6.0/icons/hands-asl-interpreting) Fontawesome glyph seem to be a universally suitable indicator for “sign language”.
- Sign languages suffer from the same Babylonian confusion like spoken and written languages. There's a lot of colour and dialects involved as well as local sign-language variants.
  - Like a person from Germany who is able to understand the German variant of the spoken german language, they are also likely to understand most of the Austrian variant, to a somewhat lesser extent though even the Swiss-German variant but not necessarily English, French Russian or Japanese. We may guess that a person who is able to understand [German german sign-language (DGS)](https://de.wikipedia.org/wiki/Deutsche_Geb%C3%A4rdensprache) will most likely be able to also understand most of [Austria-german-sign language (ÖGS)](https://de.wikipedia.org/wiki/%C3%96sterreichische_Geb%C3%A4rdensprache) but not necessarily [Varieties of the American sign language](https://en.wikipedia.org/wiki/Varieties_of_American_Sign_Language), the [French sign-language family](https://en.wikipedia.org/wiki/French_Sign_Language_family), [Russian sign-language](https://en.wikipedia.org/wiki/Russian_Sign_Language) or [Japanese sign-language](https://en.wikipedia.org/wiki/Japanese_Sign_Language_family).
  - So the spoken/written language a particular sign-language relates to closest must not to be overlooked.
- Flags are [known to be a bad representation](http://www.flagsarenotlanguages.com/blog/why-flags-do-not-represent-language/) to indicate languages. It's hard to convey language in a concise and precise way alike.
- While the technical approach to use [Locale](https://en.wikipedia.org/wiki/Locale_(computer_software))s might be appealing, most people *using* software (opposed to developing one) are unlikely to understand them easily if at all. They *may* be suitable for working with that kind of information in the backend though.
- It's a lot of information to be presented in a restricted area of screen real estate.
  - It may be suitable to use the general “sign-language translation available” indicator (see above) to be shown on the schedule overview with a rollover/tap overlay to present the details of the exact languages (of whatever kind). I personally think it's key to make that info available without having to resort to an event's detail page, since that seems inconvenient to use.
  - This info should also be available in machine-readable form, for example “FRAB-XML” so it can be show on an [Info-Beamer](https://info-beamer.com/) display which are commonly used in conjunction with Pretalx. (This might even warrant an enhanced, richer Pretalx-XML export format.)
- A language filter on the schedule page which still show *all* events, but “greys out” all events that don't match the filter would certainly help people to have an easier way to find stuff that does sport a certain language. A way to filter talks that do have sign-language translation was explicitly requested to our team. If available it would help participants and attendees of an event to have better autonomy in discovering these events. It would *also* help the support team (Infodesk) to quickly answer that question when brought up. Though, this would be helpful for any language (source or translated).
  - Event direction and video direction, especially when streaming live, will also benefit from that info being easy to spot. This might also be relevant for automated systems that obtain information from the schedule (XML).



I personally consider the [Language Turnstile Language Icon by Farhat Datta](https://competition.adesignaward.com/design.php?ID=22868) a good option to indicate language options. (In participant facing pages and event orga only pages of the Pretalx UI.)
